trading_cost.py

```diff
@@ -0,0 +1,62 @@
+"""
+A file to build cost objects.
+"""
+
+from collections import defaultdict
+
+class CurrencyCost:
+    """
+    Cost function of currency trading.
+    """
+    def __init__(self, base_ins, instruments):
+        self.base_ins = base_ins
+        self.instruments = instruments
+        self.quantity = defaultdict(int)
+        self.value = defaultdict(int)
+        self.value[base_ins] = 1
+        self.quantity[base_ins] = 1
+
+    def __call__(self, portfolio, price, liquidation):
+        if liquidation:
+            for instrument in self.instruments:
+                portfolio[-1][instrument] = (
+                    0 if instrument is not self.base_ins else 1)
+        return self.calculate(portfolio[-1], price)
+
+    def calculate(self, today, price):
+        """
+        Calculate pnl and produce cost.
+
+        We assume that our overall investment size is always 1.
+        Thus, performance of alphas is more readable because it stands for
+        ratio as well.
+        Please check wiki page for more details.
+        """
+        pnl = {instrument: 0 for instrument in self.instruments}
+        cost = {instrument: 0 for instrument in self.instruments}
+        total_value = 0
+        for instrument in self.instruments:
+            if (price[instrument][-1].buy is not None and
+                    price[instrument][-1].sell is not None):
+                position = today.get(instrument, 0)
+                middle_price = (
+                    (price[instrument][-1].buy +
+                     price[instrument][-1].sell) / 2)
+                difference = (
+                    self.quantity[instrument] - position / middle_price)
+                spread = (
+                    (price[instrument][-1].sell -
+                     price[instrument][-1].buy) / 2)
+                self.quantity[instrument] = position / middle_price
+                self.value[instrument] = position
+                cost[instrument] = abs(difference * spread)
+                pnl[instrument] = (
+                    difference * middle_price - cost[instrument])
```

Comment:
I don't think this is PnL...
This implies that any buy on some instrument will contribute to pnls of the instrument.